circuit/src/get_account.nr

```diff
@@ -6,20 +6,19 @@ struct AccountWithProof {
     balance: Field,
     codeHash: [u8; 32],
     nonce: Field,
-    state_root: [u8; proof::KEY_LENGTH],
     key: [u8; 20],
     value: [u8; proof::MAX_ACCOUNT_STATE_LENGTH],
     proof: [u8; PROOF_LEN],
     depth: Field,
 }
 
 #[oracle(get_account)]
-unconstrained fn get_account_oracle(_block_no: Field, _address: [u8; 20]) -> AccountWithProof {}
+unconstrained fn get_account_oracle(_block_no: Field, _address: [u8; 20], _state_root: [u8; proof::KEY_LENGTH]) -> AccountWithProof {}
 
-unconstrained fn get_account(block_no: Field, address: [u8; 20]) -> AccountWithProof {
-    let account = get_account_oracle(block_no, address);
+unconstrained fn get_account(block_no: Field, address: [u8; 20], state_root: [u8; proof::KEY_LENGTH]) -> AccountWithProof {
+    let account = get_account_oracle(block_no, address, state_root);
     let trie_proof = proof::TrieProof { key: account.key, proof: account.proof, depth: account.depth, value: account.value };
-    assert(trie_proof.verify_state_root(account.state_root));
+    assert(trie_proof.verify_state_root(state_root));
```

Comment:
Just realised we are using asserts in unconstrained that seems to be wrong.
Should not work (somehow tests pass, not sure why).

Reply:
Will be done in a separate ticket https://github.com/orgs/zkworks-xyz/projects/5/views/1?pane=issue&itemId=49181566